Build yetus image on demand

rene opened the pull request:
# Description

This PR introduces two things:

1. The source code for the Yetus image we are using in our GH actions
2. A new GH workflow to build and push this image on demand

Since we don't update Yetus image very often, the idea is to provide the action to build it only when some update is needed. By using this approach we don't need to keep a new/separated repository (e.g. lf-edge/eve-yetus) only for this image.

The version (TAG) is marked as the following:

`0.15.1-eve-1`

where:

`0.15.1` (Yetus base image version) + `eve-1` (represents the version of our image, future images will be eve-2, eve-3, etc)

## How to test and validate this PR

Run the workflow manually (since it runs on demand) and check that it works.

PS: This PR was tested on my repo: https://github.com/rene/eve/actions/runs/19572770766/job/56050134859

Images for amd64 + arm64 are published.

## Changelog notes

No user-facing changes.

## PR Backports

The Yetus image is the same across the other branches (in their actions), so there is no need to backport this PR.

## Checklist

- [x] I've provided a proper description
- [x] I've added the proper documentation
- [ ] I've tested my PR on amd64 device
- [ ] I've tested my PR on arm64 device
- [x] I've written the test verification instructions
- [x] I've set the proper labels to this PR
- [x] I've checked the boxes above, or I've provided a good reason why I didn't
  check them.

shjala commented on .github/workflows/buildyetusondemand.yml:
```diff
@@ -0,0 +1,63 @@
+      - name: Clean up
+        run: |
+          docker system prune -f -a || :
+          docker rm -f $(docker ps -aq) && docker volume rm -f $(docker volume ls -q) || :
```
@rene For this one the recommended way is to use [action permission monitor ](https://github.com/GitHubSecurityLab/actions-permissions)  (either here temporary or in your own repo if you can run actions) and find out what is the minimum needed perm.

rene commented on .github/workflows/buildyetusondemand.yml:
```diff
@@ -0,0 +1,62 @@
+        with:
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          ref: ${{ github.event.pull_request.head.ref }}
+          fetch-depth: 0
```
yes, this is fixed already @shjala .

rene commented:
> > @eriknordmark , @rucoder and other reviewers, it's been 3 weeks since this PR is ready, I've addressed all comments. Appreciate your final approval....
> 
> I'm wairting for some comments from @shjala about the issues flagged by zizimor (I think I tagged him on this earlier).

Which errors @eriknordmark ? I've fixed all zizimor errors and the workflow is passing.
